tests/test_integration.py

```diff
@@ -0,0 +1,91 @@
+'14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26',
+'27', '28', '29', '30', '31']} era5_orography_2008_hourly.nc"""
+
+# without --levels surface, geopotential calls pressure level data
+call3 = """era5cli hourly --variables geopotential --startyear 2008 --dryrun"""
+
+result3 = """reanalysis-era5-pressure-levels {'variable': 'geopotential',
+'year': 2008, 'month': ['01', '02', '03', '04', '05', '06', '07', '08', '09',
+'10', '11', '12'], 'time': ['00:00', '01:00', '02:00', '03:00', '04:00',
+'05:00', '06:00', '07:00', '08:00', '09:00', '10:00', '11:00', '12:00',
+'13:00', '14:00', '15:00', '16:00', '17:00', '18:00', '19:00', '20:00',
+'21:00', '22:00', '23:00'], 'format': 'netcdf', 'pressure_level': [1, 2, 3, 5,
+7, 10, 20, 30, 50, 70, 100, 125, 150, 175, 200, 225, 250, 300, 350, 400, 450,
+500, 550, 600, 650, 700, 750, 775, 800, 825, 850, 875, 900, 925, 950, 975,
+1000], 'product_type': 'reanalysis', 'day': ['01', '02', '03', '04', '05',
+'06', '07', '08', '09', '10', '11', '12', '13', '14', '15', '16', '17', '18',
+'19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31']}
+era5_geopotential_2008_hourly.nc"""
+
+# preliminary-back-extension is combined with monthly-means
+call4 = """era5cli monthly --variables temperature --startyear 1960 --prelimbe
+--dryrun"""
+
+result4 = """reanalysis-era5-pressure-levels-monthly-means-preliminary-back-extension
+{'variable': 'temperature', 'year': 1960, 'month': ['01', '02', '03', '04',
+'05', '06', '07', '08', '09', '10', '11', '12'], 'time': ['00:00'], 'format':
+'netcdf', 'pressure_level': [1, 2, 3, 5, 7, 10, 20, 30, 50, 70, 100, 125, 150,
+175, 200, 225, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 775, 800,
+825, 850, 875, 900, 925, 950, 975, 1000], 'product_type':
+'reanalysis-monthly-means-of-daily-means'} era5_temperature_1960_monthly.nc"""
+
+# era5-Land is combined with monthly means
+call5 = """era5cli monthly --variables snow_cover --startyear 2008 --land
+--dryrun"""
+
+result5 = """reanalysis-era5-land-monthly-means {'variable': 'snow_cover',
+'year': 2008, 'month': ['01', '02', '03', '04', '05', '06', '07', '08', '09',
+'10', '11', '12'], 'time': ['00:00'], 'format': 'netcdf', 'product_type':
+'monthly_averaged_reanalysis'} era5-land_snow_cover_2008_monthly.nc"""
+
+call_result = [
+    (call1, result1),
+    (call2, result2),
+    (call3, result3),
+    (call4, result4),
+    (call5, result5)
+    ]
+ids = [call[0] for call in call_result]
+
+
+@pytest.mark.parametrize("call,result", call_result, ids=ids)
+def test_main(call, result, capsys):
+    call = call.split()
+    result = result.replace('\n', ' ') + '\n'
+    # until the actual fetch is monkeypatched, make sure the tests are dryruns
+    if '--dryrun' not in call:
+        pytest.fail('call must be a dryrun')
+    cli.main(call)
+    captured = capsys.readouterr().out
+    assert result == captured
```

Comment:
Some calls generate warning log messages like
* "WARNING  root:fetch.py:369 The variable 'orography' has been deprecated by CDS. Use `--variables geopotential --levels surface` going forward. The current query has been changed accordingly."
* "WARNING:root:The variable name 'geopotential' is ambiguous. Getting variable from surface level data. To get the pressure variable instead, omit `--levels surface` or `levels=['surface']` from the request."

It would be nice to also test those using [caplog fixture](https://docs.pytest.org/en/6.2.x/logging.html#caplog-fixture) and an extra tuple member to call_result var.